### DISCUSSION TITLE
Yarn throws "Segmentation Fault" whenever updating a Node dependency

### DISCUSSION THREAD

**Discussion by Wassim-AB** [renovatebot/renovate#23998] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

5.0.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Hello :)

Recently we have been getting "Segmentation Fault" errors when Renovate was trying to refresh the ```yarn.lock``` whenever it was updating JS dependencies. You can find of the log from Whitesource Renovate down below.

I've checked the CPU, Memory and Disk usage of our Renovate Container (deployed within a Kubernetes Cluster) and everything is fine, consumption is far below the limits that we gave to the container.

So we figured out that maybe it's because we still use Yarn v1, so we decided to upgrade it. We did the following :

1) We put the yarn binary in the repository under .yarn/releases/yarn-3-6-2.cjs
2) We created a .yarnrc.yml file with the following content :
  ```yaml
  nodeLinker: node-modules
  
  yarnPath: .yarn/releases/yarn-3.6.2.cjs
  ```

Renovate successfully rebased the PRs, but now we have this error : 
```
No /opt/containerbase/tools/corepack/0.19.0/bin defined - aborting
```

We are unable to reproduce this errors locally, or when freshly cloning the repository. I've searched through the Yarn and Renovate issues but I couldn't find anything relevant. Do you have any clue ?

Thank you !

### Relevant debug logs

<details><summary>Logs</summary>

```json
{
   "name":"renovate",
   "hostname":"xxx",
   "pid":7,
   "level":40,
   "logContext":"sM2NDKOkM",
   "repository":"xxx",
   "branch":"renovate/chai-4.x",
   "artifactErrors":[
      {
         "lockFile":"yarn.lock",
         "stderr":"Segmentation fault\n"
      }
   ],
   "msg":"artifactErrors",
   "time":"2023-08-21T13:16:13.807Z",
   "v":0
}
```
</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by viceice** [comment]
you should also set the packageManager option inside the package.json

**Reply by Wassim-AB** [reply]
Hey, forgot to mention it but we also added it with this line :
```json
"packageManager": "yarn@3.6.2"
```

**Reply by viceice** [reply]
ok, please check the logs, there should be some debug messages from the yarn exec call. add those to the logs section above

**Reply by Wassim-AB** [reply]
Hey, I changed the LOG_LEVEL in order to give you the logs you asked for, but apparently, rebooting it just made it worked again for the PRs that were causing the issue :D

I'll keep you in touch if it ever happens again tomorrow, otherwise I'll close the discussion.

Thank you for your time !
